protocol-designer/src/molecules/DropdownStepFormField/index.tsx

```diff
@@ -31,16 +37,43 @@ export function DropdownStepFormField(
     padding = `0 ${SPACING.spacing16}`,
     width = '17.5rem',
     onFieldFocus,
+    onEnter,
+    onExit,
     onFieldBlur,
+    name: fieldName,
   } = props
-  const { t } = useTranslation('tooltip')
+  const { t, i18n } = useTranslation(['tooltip', 'application'])
+  const dispatch = useDispatch()
   const availableOptionId = options.find(opt => opt.value === value)
-
-  useEffect(() => {
-    if (options.length === 1) {
-      updateValue(options[0].value)
+  const handleSelection = (value: string): void => {
+    let text = t('application:selected')
+    if (fieldName === 'newLocation') {
+      text = i18n.format(t('application:location'), 'capitalize')
+    } else if (fieldName === 'aspirate_labware') {
+      text = t('application:source')
+    } else if (fieldName === 'dispense_labware') {
+      text = t('application:dest')
+    }
```

Comment:
maybe not necessary but nit to avoid `let`s: 
```suggestion
const fieldMappedToTranslationKey = {
  'newLocation': 'location',
  'aspirate_labware': 'source',
  'dispense_labware': 'dest'
}
const text = i18n.format(t(`application:`${fieldMappedToTranslationKey[fieldName] ?? 'selected'}`, 'capitalize')
```